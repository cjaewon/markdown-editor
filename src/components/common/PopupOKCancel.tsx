import * as React from 'react';
import styled from 'styled-components';
import PopupBase from './PopupBase';
import Button from './Button';
import palette from '../../lib/styles/palette';

const PopupOKCancelBlock = styled.div`
  h3 {
    margin: 0;
    font-size: 1.5rem;
    color: ${palette.gray8};
    line-height: 1.5;
    font-weight: bold;
  }
  .message {
    line-height: 1.5;
    font-size: 1rem;
    color: ${palette.gray7};
    margin-top: 1rem;
    margin-bottom: 1rem;
    white-space: pre-wrap;
  }
  .button-area {
    margin-top: 2rem;
    display: flex;
    justify-content: flex-end;
    button + button {
      margin-left: 0.75rem;
    }
  }
`;

export interface PopupOKCancelProps {
  visible: boolean;
  title?: string;
  onConfirm?: () => any;
  onCancel?: () => any;
  children: React.ReactNode;
}

const PopupOKCancel: React.FC<PopupOKCancelProps> = ({
  visible,
  title,
  children,
  onConfirm,
  onCancel,
}) => {
  return (
    <PopupBase visible={visible}>
      <PopupOKCancelBlock>
        {title && <h3>{title}</h3>}
        <div className="message">{children}</div>
        <div className="button-area">
          {onCancel && (
            <Button color="lightGray" onClick={onCancel}>
              취소
            </Button>
          )}
          <Button onClick={onConfirm}>확인</Button>
        </div>
      </PopupOKCancelBlock>
    </PopupBase>
  );
};

export default PopupOKCancel;
