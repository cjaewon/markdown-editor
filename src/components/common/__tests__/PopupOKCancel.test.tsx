import * as React from 'react';
import { render, fireEvent } from '@testing-library/react';
import PopupOKCancel, { PopupOKCancelProps } from '../PopupOKCancel';

describe('PopupOKCancel', () => {
  const setup = (props: Partial<PopupOKCancelProps> = {}) => {
    const initialProps: PopupOKCancelProps = {
      visible: true,
      title: '제목',
      children: '내용',
    };
    const utils = render(<PopupOKCancel {...initialProps} {...props} />);
    return {
      ...utils,
    };
  };
  it('renders properly', () => {
    const utils = setup();
    utils.getByText('제목');
    utils.getByText('내용');
  });
  it('matches snapshot', () => {
    const { container } = setup();
    expect(container).toMatchSnapshot();
  });
  describe('visible property works properly', () => {
    it('false', () => {
      const utils = setup({ visible: false });
      const title = utils.queryByText('제목');
      expect(title).toBe(null);
    });
    it('true', () => {
      const utils = setup({ visible: true });
      utils.getByText('제목');
    });
  });
  it('button works properly', () => {
    const [onConfirm, onCancel] = [jest.fn(), jest.fn()];
    const utils = setup({
      onConfirm,
      onCancel,
    });
    const confirm = utils.getByText('확인');
    const cancel = utils.getByText('취소');
    fireEvent.click(confirm);
    expect(onConfirm).toBeCalled();
    fireEvent.click(cancel);
    expect(onCancel).toBeCalled();
  });
});
