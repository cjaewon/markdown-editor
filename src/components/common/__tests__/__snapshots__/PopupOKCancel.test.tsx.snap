// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PopupOKCancel matches snapshot 1`] = `
<div>
  <div
    class="sc-bdVaJa kNCPvv"
  />
  <div
    class="sc-bwzfXH fIoYbF"
  >
    <div
      class="sc-htpNat gFFQdP"
    >
      <div
        class="sc-ifAKCX lbyxTT"
      >
        <h3>
          제목
        </h3>
        <div
          class="message"
        >
          내용
        </div>
        <div
          class="button-area"
        >
          <button
            class="sc-bxivhb jvTLUf"
            color="teal"
          >
            확인
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
